Бездольный Н.А.

gooverdian opened the pull request:
Некоторые вещи не такие, как в макете, например, расстояния, или размеры шрифтов, или цвета.
Очень надеюсь, что попиксельная подгонка не является частью ДЗ.

gooverdian committed:
Общая структура и готовый футер

gooverdian committed:
Изображения и один из разделов

gooverdian committed:
Клиенты и новости

gooverdian committed:
Портфолио и команда

gooverdian committed:
Исправления

Добавлена пустая строка в конце во все файлы
Исправлен HTML элемент во многих местах

sergdenisov reviewed (changes requested):
Я же говорил про стайлгайд — https://google.github.io/styleguide/htmlcssguide.html#Indentation. Должно быть 2-4 отступа везде, а то читать совсем нереально код с таким форматированием.

gooverdian committed:
Исправление форматирования

Все отступы переделаны в 4 пробела.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        </p>
+                    </div>
+                </div>
+                <section class="slidable intro intro_small">
```
Эта секция должна лежать либо отдельно (не внутри родительской секции), или то что лежит выше (заголовок и текст) нужно тоже положить в секцию.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                    </div>
+                </div>
+                <section class="slidable intro intro_small">
+                    <a href="#" class="slidable__slider">
```
Семантически это не ссылка, а `<button>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <h5 class="article__heading">
+                                    Comic
+                                </h5>
+                                <blockquote class="quote article__quote">
```
Это же не цитата, а просто кусок текста — `<p>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                        A million ways to die in the wild wild wild west
+                                    </p>
+                                </blockquote>
+                                <!-- Картинки вместо классов CSS использованы специально, потому как это 
```
Такие комменты не нужны в целом. Тем более, я помню что писал в соседнем пулл-реквесте =).

gooverdian replied:
Этот коммент был написан до того, как я посмотрел соседний пул-реквест

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <span class="icon icon_standalone icon_chevron-right"></span>
+                    </a>
+                    <div class="row align-items-stretch">
+                        <div class="col">
```
`<article>`: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/article

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                </blockquote>
+                                <!-- Картинки вместо классов CSS использованы специально, потому как это 
+                                         юзерпики и при генерации контента будут в HTML, а не в CSS -->
+                                <footer class="article__footer">
```
Ну, тогда логично было бы и заголовок обернуть в `<header>` =). Но мне кажется эти теги оверхэдом для статьи в данном случае, лучше без них.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                        <img class="userpic__image" src="img/usr/2.png" alt="User 2" />
+                                        <div class="userpic__description">
+                                            Mason Johnston
+                                            <br>
```
Лучше не использовать `<br>`, а положить в отдельные элементы, поскольку этот контент, скорее всего, приходит с бэкенда по задумке и ты не знаешь что там может быть внутри.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <div class="col-11">
+                            <h2 class="title title_level-2">Clients</h2>
+                            <div class="list list_inline slidable">
+                                <!-- Не ul, потому что внутри слайдер -->
```
1. Опять же, коммент не нужен и я все помню.
2. Не вижу причин не сделать `<ul>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+
+            <section class="container outstanding">
+                <div class="row justify-content-center">
+                    <div class="col-9">
```
Лучше всегда четное количество использовать, 8 или 10 в данном случае, иначе у тебя отступы получаются не размером с целую колонку, а только с ее часть.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+            <section class="substrate">
+                <div class="container">
+                    <div class="row justify-content-center clients">
+                        <div class="col-11">
```
Аналогично — 10 или 12.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                    <span class="icon icon_standalone icon_chevron-right"></span>
+                                </a>
+                                <span class="list__item">
+                                    <span class="clients__item clients_client-1"></span>
```
Картинки лучше через `<img>`, так как это динамический контент.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                            <h2 class="title title_level-2">Clients</h2>
+                            <div class="list list_inline slidable">
+                                <!-- Не ul, потому что внутри слайдер -->
+                                <a href="#" class="slidable__slider">
```
Тоже `<button>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <a href="#" class="portfolio__switch"></a>
+                    </li>
+                    <li class="list__item">
+                        <a href="#" class="portfolio__switch portfolio_switch-active"></a>
```
Нельзя модификатор для блока и его же элемент на одном HTML-элементе.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                </ul>
+                <ul class="portfolio__switcher list list_inline list_close">
+                    <li class="list__item">
+                        <a href="#" class="portfolio__switch"> </a>
```
`<button>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        </li>
+                    </ul>
+                </div>
+                <ul class="portfolio__list list list_inline list_close">
```
Тут ничего не должно выходить за границы страницы визуально, это же слайдер.

<img width="1440" alt="screen shot 2018-01-05 at 20 26 33" src="https://user-images.githubusercontent.com/5359222/34620525-cf190e72-f256-11e7-982a-e6c4d27b6a29.png">

gooverdian replied:
Вот пакость =) Это уже вопрос кросс-браузерности, в Мозилле всё ок, я поправлю для остальных
  
Нет, соврал, и там не ок - видимо, какое-то из исправлений добавило этот баг )

sergdenisov replied:
Ну это не похоже на проблемы браузеров. И пользоваться в первую очередь лучше Хромом =).

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+
+            <section class="container outstanding">
+                <div class="row justify-content-center">
+                    <div class="col-9">
```
Опять же, 8 или 10.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                </div>
+                <div class="row members justify-content-around">
+                    <div class="col-2 members__item">
+                        <span class="squarehead squarehead_item-1"></span>
```
Опять же, контент динамический скорее всего будет и делать на каждый элемент в CSS `_item-xxx` у тебя не будет возможности в реале.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+            <section class="facts">
+                <div class="container outstanding outstanding_more">
+                    <div class="row justify-content-center">
+                        <div class="col-9">
```
8-10.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <div class="row facts__row">
+                                    <div class="col facts__item">
+                                        <h5 class="facts__number">110</h5>
+                                        <div>successful projects</div>
```
Отображением регистра букв (большие/маленькие) лучше всегда заниматься только в CSS.

sergdenisov replied:
И тут лучше не `<div>`, а тоже что-то типа `<h5>`-`<h6>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <h5 class="text text_muted">
+                            Travel / 21st March, 2014
+                        </h5>
+                        <h2 class="title title_level-2-thin">
```
Нужно разбивать на 2 модификатора: `title_level-2` и `title_thin` и вешать оба.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                    </div>
+                    <img class="news__picture" src="img/news/news1.png" alt="News 1" />
+                </div>
+                <div class="news__item news_reverse">
```
Опять же, нельзя элемент блока и модификатор этого же блока на один HTML-элемент вешать.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+
+            <section class="substrate">
+                <div class="container slidable headlines">
+                    <a href="#" class="slidable__slider">
```
`<button>`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <span class="icon icon_standalone icon_chevron-right"></span>
+                    </a>
+                    <div class="row">
+                        <div class="col headlines__item">
```
`<article>` наверное лучше подойдет тут.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                Happiness cannot be traveled to, owned, earned or worn. It is the spiritual
+                                experience of living every minute with love, grace &amp; gratitude.
+                            </p>
+                            <footer class="quote__footer quote_twitter">
```
Модификатор от блока с элементом опять же.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                        <label for="email-input" class="field__label">
+                                            Email
+                                        </label>
+                                        <input id="email-input" class="field__input" type="text" />
```
Лучше `<input>` класть прям внутрь `<label>` по возможности, тогда не нужны будут атрибуты `for` и `id`.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <textarea id="message-input" class="field__input"></textarea>
+                            </div>
+                            <div class="field">
+                                <input class="field__button" type="submit" value="Send ►" />
```
Лучше `<button>` всегда использовать для кнопок.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        <h5 class="footer__heading">
+                            Links
+                        </h5>
+                        <ul class="list">
```
Еще хорошей семантикой считается в `<nav>` оборачивать вещи, связанные с навигацией.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                <div class="row align-items-center footer_light footer_divider">
+                    <div class="col-6">
+                        One morning when Gregor Samsa woke from troubled dreams, he found
+                        <br>
```
Опять же, лучше без `<br>`.

sergdenisov commented on less/article.less:
```diff
@@ -0,0 +1,44 @@
+}
+
+    .article__heading {
+        font-size: 1em;
```
Лучше не использовать `em`, если точно не уверен, что они тебе нужны =). Обычно хватает `px`.

gooverdian replied:
Здесь использован 1em, что равносильно font-size: inherit; и написан он именно для этой цели, лучше написать inherit?

sergdenisov replied:
Это совсем не равносильно =). Тем более, размер шрифта и так наследуется. В данном случае вообще тогда не ясно, зачем его тут указывать? У блока выше он же уже указан.

gooverdian replied:
В HTML этот элемент лежит в каком-то тэге, например h5, как здесь.. как мне указать, что я хочу, чтобы размер шрифта совпадал с остальным текстом, независимо от тега?

sergdenisov replied:
Ну тогда ты просто задаешь размер шрифта, который тебе нужен, в `px`.

sergdenisov commented on less/article.less:
```diff
@@ -0,0 +1,44 @@
+    }
+
+    .article__quote {
+        font-size: 1.5rem;
```
И особенно не мешать все вместе: `em`, `px`, `rem` для одних и тех же целей.

sergdenisov replied:
Это касается вообще всех файлов. `em`/`rem` нужно использовать только когда есть реальная логика, которая завязана на размер шрифта.

sergdenisov commented on less/article.less:
```diff
@@ -0,0 +1,44 @@
+    .article__quote {
+        font-size: 1.5rem;
+        line-height: 1.5;
+        letter-spacing: .07em;
```
Вот тут кстати подходящее место для `em`.

sergdenisov commented on less/article.less:
```diff
@@ -0,0 +1,44 @@
+        &:before {
+            content: " ";
+            position: absolute;
+            width: 5em;
```
Опять же, лучше пиксели.

sergdenisov commented on less/field.less:
```diff
@@ -0,0 +1,72 @@
+        border-bottom: 1px solid @color-footer-geometry;
+    }
+
+    .field__label,
```
Лучше для каждой сущность писать отдельный селектор. Когда у тебя селекторы `.field__label` размазаны по файлу, понимать код тяжело. Если свойства дублируются, то можно использовать миксины.

sergdenisov commented on less/field.less:
```diff
@@ -0,0 +1,72 @@
+        display: block;
+        width: 100%;
+
+        textarea& {
```
Селекторы по названиям тегов нельзя.

sergdenisov commented on less/field.less:
```diff
@@ -0,0 +1,72 @@
+    .field__input {
+        color: @color-footer-heading-text;
+        display: block;
+        width: 100%;
```
Это не нужно для `display: block`, он и так на всю ширину.

gooverdian replied:
https://jsfiddle.net/owwthj20/1/

sergdenisov commented on less/field.less:
```diff
@@ -0,0 +1,72 @@
+    display: block;
+}
+
+    .field__label {
```
Вообще, судя по макетам, это должен быть не лейбл, а просто `placeholder` https://developer.mozilla.org/en-US/docs/Web/CSS/::placeholder.

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,533 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                        </ul>
+                    </section>
+                </div>
+                <div class="row align-items-center footer_light footer_divider">
```
Нельзя использовать модификаторы без основного блока.

sergdenisov commented on less/footer.less:
```diff
@@ -0,0 +1,45 @@
+    padding: 50px 0 35px;
+}
+
+    .footer_divider {
```
Возможно это должно быть элементом? Или название не подходящее для модификатора.

sergdenisov commented on less/footer.less:
```diff
@@ -0,0 +1,45 @@
+        color: @color-footer-light-text;
+    } 	
+
+    .footer_divider::before {
```
Лучше перенести выше, к самому модификатору.

sergdenisov commented on less/footer.less:
```diff
@@ -0,0 +1,45 @@
+        left: 0;
+        right: 0;
+        content: ' ';
+        display: block;
```
Можно не добавлять по идее, и так будет.

sergdenisov commented on less/footer.less:
```diff
@@ -0,0 +1,45 @@
+        line-height: 20px;
+    }
+
+    h4.footer__heading {
```
Завязки на теги быть не должно.

sergdenisov commented on less/link.less:
```diff
@@ -2,9 +2,28 @@
+        }
+    }
+
+    .link_footer-light {
```
Нужно разделить на 2 модификатора.

sergdenisov commented on less/list.less:
```diff
@@ -0,0 +1,37 @@
+        display: block;
+    }
+
+    .list_pulled-right {
```
Лучше называть по смысле, а не по стилям, которые модификатор применяет.

sergdenisov commented on less/list.less:
```diff
@@ -0,0 +1,37 @@
+        }
+    }
+
+    .list_close {
```
Возможно лучше сделать модификатор для элемента, а не для блока, раз тут всего 1 элемент. Выше, кстати, тоже.

gooverdian replied:
Лучше, чтобы у всех айтемов в списке был дополнительный класс?
Или класс айтема нужно присвоить самому списку?
Ни то ни другое мне не кажется логичным, и я не могу понять, что требуется
Смысл этого модификатора - все айтемы списка располагаются близко друг к другу..

sergdenisov replied:
Тут все просто — если модификатор модифицирует блок, то это модификатор блока. Если модификатор модифицирует элемент, то это модификатор элемента. Ты здесь сделал модификатор для блока, который нужен только для модификации элементов.

gooverdian replied:
То есть делаем этот модификатор list__item_close и вешаем на каждый элемент списка? Так правильнее?

sergdenisov replied:
Ну, на самом деле, строгих правил нет. Но я стараюсь придерживаться такой логики, что если мы модифицируем только 1 элемент с помощью модификатора, то модификатор должен быть у этого элемента. Поэтому ответ — да.

gooverdian replied:
Это несколько расходится у меня в голове с тем, что требуется сделать, такая логика подразумевает, что у какого-то айтема будет такой модификатор, а у какого-то нет в рамках одного списка. А я этим модификатором хочу показать, что у этого **списка все** айтемы расположены рядом. Я в общем-то поправлю, это недолго, но в данном конкретном случае, я так и не согласен.

> Если модификатор модифицирует элемент, то это модификатор элемента.

Этот модификатор модифицирует **блок** так, чтобы **все** элементы этого блока . . .

sergdenisov replied:
Согласен, объяснение звучит логично. В данном случае можно сделать как ты говоришь, но тогда я бы переименовал модификатор, потому что в он звучит как модификатор элемента, а не списка. Список получается узкий, поэтому я назвал бы как-то типа `.list_narrow`.

sergdenisov commented on less/members.less:
```diff
@@ -0,0 +1,20 @@
+@import "squarehead";
```
Блоки должны быть независимы, один блок не должен ничего знать о другом блоке.

gooverdian replied:
Причина этого в том, что squearehead не существует без member в рамках макета и разметки. Но да, я убрал

sergdenisov replied:
В верстке он лежит вроде просто как блок, все ок.

sergdenisov commented on less/outstanding.less:
```diff
@@ -0,0 +1,9 @@
+.outstanding {
```
Имена блоков и элементов по логике лучше делать существительными.

sergdenisov commented on less/portfolio.less:
```diff
@@ -0,0 +1,55 @@
+}
+
+    .portfolio__links {
+        font-size: 1.05em;
```
Не нужно так заморачиваться с размером =). Тем более, там наверняка не получается целое число, что плохо.

sergdenisov commented on less/portfolio.less:
```diff
@@ -0,0 +1,55 @@
+    .portfolio__list {
+        text-align: left;
+        margin-bottom: 60px;
+        height: 359px;
```
360.

sergdenisov commented on less/portfolio.less:
```diff
@@ -0,0 +1,55 @@
+    }
+
+    .portfolio__image {
+        height: 359px;
```
100%.

sergdenisov commented on less/portfolio.less:
```diff
@@ -0,0 +1,55 @@
+        height: 359px;
+    }
+
+    .portfolio__switch {
```
`switcher` тогда лучше.

sergdenisov commented on less/portfolio.less:
```diff
@@ -0,0 +1,55 @@
+        }
+    }
+
+    .portfolio_switch-active {
```
Лучше прилагательные использовать в названиях модификаторов.

sergdenisov commented on less/slidable.less:
```diff
@@ -0,0 +1,20 @@
+@import "variables";
+
+.slidable {
```
Лучше `.slider`.

sergdenisov commented on less/slidable.less:
```diff
@@ -0,0 +1,20 @@
+    position: relative;
+}
+
+    .slidable__slider {
```
Что-то типа `.slider__switcher`.

sergdenisov commented on less/title.less:
```diff
@@ -26,6 +27,17 @@
         }
     }
 
+    .title_level-2-thin {
```
На 2 модификатор надо разбивать.

sergdenisov commented on less/userpic.less:
```diff
@@ -0,0 +1,29 @@
+    }
+
+    .userpic_standalone {
+        justify-content: center !important;
```
Нужно сделать без `!important`.

sergdenisov commented on less/variables.less:
```diff
@@ -1 +1,12 @@
-@color-text: #212121;
+@color-main-text: #212121;
+@color-main-background: #fff;
```
Лучше называть наоборот, типа `@color-background-main`. Потому что там удобней сортировать переменные по типу.

gooverdian committed:
Исправление Featured Articles

gooverdian committed:
Исправление Clients

gooverdian committed:
Исправление Portfolio

gooverdian committed:
Исправление Members

gooverdian committed:
Исправление Facts

gooverdian committed:
Исправление News

gooverdian committed:
Исправление Headlines и Quotes

gooverdian committed:
Исправление Footer

gooverdian committed:
Исправление Less

sergdenisov commented on less/list.less:
```diff
@@ -0,0 +1,37 @@
+        float: right;
+    }
+
+    .list_inline {
```
Лучше, кстати, назвать как-то типа `.list_horizontal`.

gooverdian committed:
Исправление List

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,527 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <h5 class="article__heading">
+                                    Comic
+                                </h5>
+                                <p class="article__quote quote quote__text">
```
Нельзя сразу блок и его элемент на одной HTML-ноде.

gooverdian replied:
Ой, это я поправил в первом, и про два других забыл

sergdenisov commented on index.html:
```diff
@@ -110,8 +110,527 @@ <h3 class="title title_level-3">An Amazing Feature</h3>
+                                <h5 class="article__heading">
+                                    Lifestyle
+                                </h5>
+                                <p class="article__quote quote quote__text">
```
Тут тоже самое.

sergdenisov commented on less/all.less:
```diff
@@ -1,11 +1,27 @@
 @import "banner";
 @import "link";
 @import "nav";
+@import "text";
```
Точняк, я у себя забыл заимпортить =).

sergdenisov commented on less/clients.less:
```diff
@@ -0,0 +1,11 @@
+        width: 100px;
+        height: 100px;
+        margin: 15px;
+    }
```
Пустая строка должна быть https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline.

sergdenisov commented on less/field.less:
```diff
@@ -0,0 +1,46 @@
+    border-bottom: 1px solid @color-geometry-footer;
+}
+
+    .field-controls() {
```
Лучше в самый верх файла вынести.

sergdenisov commented on less/field.less:
```diff
@@ -0,0 +1,46 @@
+            color: @color-text-footer-light;
+            text-decoration: underline;
+        }
+    }
```
New line.

sergdenisov commented on less/headlines.less:
```diff
@@ -0,0 +1,22 @@
+    }
+
+    .headlines__heading {
+        font-size: 1.2rem;
```
`rem` лучше везде убрать, поскольку у тебя тут нет никакой завязки на изменение размера шрифта root-элемента при каких-то условиях.

sergdenisov commented on less/link.less:
```diff
@@ -2,9 +2,28 @@
+        color: @color-text-footer-light;
+        text-transform: uppercase;
+        font-weight: 500;
+        letter-spacing: -.05em;
```
Вкусовщина, но мне кажется наглядней писать `-0.05em` (править не обязательно).

sergdenisov commented on less/list.less:
```diff
@@ -0,0 +1,36 @@
+
+        .list__item {
+            display: inline-block;
+            margin: 0 1em;
```
Тут тоже я бы в `px` задал.

sergdenisov commented on less/mixins.less:
```diff
@@ -2,3 +2,16 @@
+    left: 0;
+    background: @color-accent-main;
+    margin: 0 auto;
+}
```
New line.

sergdenisov commented on less/news.less:
```diff
@@ -0,0 +1,33 @@
+
+    .news__item_reverse {
+
+        .news__picture {
```
Так нельзя по БЭМу — у тебя один элемент модифицирует другой элемент, у элементов не должно быть зависимости друг от друга, только от основного блока.

sergdenisov commented on less/userpic.less:
```diff
@@ -0,0 +1,29 @@
+        border-radius: @userpic-master-width;
+        overflow: hidden;
+        line-height: 1;
+        margin-right: 1em;
```
Если явно не задаешь сущности размер шрифта, то лучше не делать завязку на `em`, поскольку у тебя получается что блок/элемент зависит от того, куда его вложишь.

gooverdian committed:
Исправления Less

Разделение блока news, добавление потерянных пустых строк,
переход от em и rem к px во многих местах

gooverdian committed:
Последняя потерянная строка

Да куда они деваются..?
